torch/_masked/__init__.py

```diff
@@ -512,6 +512,95 @@ def _apply(a):
     return result.coalesce()
 
 
+def _sparse_coo_scatter_reduction_helper(op,
+                                         mask_input: Tensor,
+                                         dims: Tuple[int, ...],
+                                         keepdim: bool,
+                                         dtype: Optional[DType] = None) -> Tensor:
+    reduce = op.__name__
+    assert reduce in {'sum', 'prod', 'mean', 'amax', 'amin'}, \
+        "op must be one of torch.sum, torch.prod, torch.mean, torch.amax or torch.amin"
```

Comment:
Also include the value that was actually received. In general I consider this a ValueError.

```
valid_reductions = ['sum', 'prod', ..., 'amin']
if op not in valid_reductions:
    throw ValueError(f"op must be one of {' '.join(valid_reductions}}, but got {op.__name__} instead")
```